pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -501,91 +509,127 @@ protected void validateDestinationOwnership(DestinationName fqdn, boolean author
-                                localCluster, namespace.toString());
-                        log.warn(msg);
-                        validationFuture.completeExceptionally(new RestException(Status.PRECONDITION_FAILED, msg));
-                    } else if (!policies.replication_clusters.contains(localCluster)) {
-                        String msg = String.format(
-                                "Global namespace missing local cluster name in replication list : local_cluster=%s ns=%s repl_clusters=%s",
-                                localCluster, namespace.toString(), policies.replication_clusters);
-
-                        log.warn(msg);
-                        // TODO: when we have a fail-over policy defined, we should find the next cluster in the
-                        // replication
-                        // clusters to re-direct the request to
-                        validationFuture.completeExceptionally(new RestException(Status.PRECONDITION_FAILED, msg));
-                    } else {
-                        validationFuture.complete(null);
+        if (!namespace.isGlobal()) {
+            return CompletableFuture.completedFuture(null);
+        }
+        final CompletableFuture<ClusterData> validationFuture = new CompletableFuture<>();
+        final String localCluster = pulsarService.getConfiguration().getClusterName();
+        final String path = AdminResource.path(POLICIES, namespace.getProperty(), namespace.getCluster(),
+                namespace.getLocalName());
+
+        pulsarService.getConfigurationCache().policiesCache().getAsync(path).thenAccept(policiesResult -> {
+            if (policiesResult.isPresent()) {
+                Policies policies = policiesResult.get();
+                if (policies.replication_clusters.isEmpty()) {
+                    String msg = String.format(
+                            "Global namespace does not have any clusters configured : local_cluster=%s ns=%s",
+                            localCluster, namespace.toString());
+                    log.warn(msg);
+                    validationFuture.completeExceptionally(new RestException(Status.PRECONDITION_FAILED, msg));
+                } else if (!policies.replication_clusters.contains(localCluster)) {
+                    ClusterData ownerPeerCluster = getOwnerFromPeerClusterList(pulsarService,
+                            policies.replication_clusters);
+                    if (ownerPeerCluster != null) {
+                        // found a peer that own this namespace
+                        validationFuture.complete(ownerPeerCluster);
+                        return;
                     }
+                    String msg = String.format(
+                            "Global namespace missing local cluster name in replication list : local_cluster=%s ns=%s repl_clusters=%s",
+                            localCluster, namespace.toString(), policies.replication_clusters);
 
+                    log.warn(msg);
```

Comment:
Why don't you use `log.warn("Global namespace missing local cluster name in replication list : local_cluster=%s ns={} repl_clusters={}", localCluster, namespace.toString()` like other logs ?

Reply:
actually this is [existing logging already present into master](https://github.com/apache/incubator-pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java?utf8=%E2%9C%93#L560). The reason of creating string `msg` is: we can use it while logging at line#584 and while creating exception-message at line#585. If we don't do it then probably we will create duplicate info-msg multiple times.

Reply:
Ok, I see